verilog/analysis/extractors.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file defines helper functions to gather the descriptions of all the
+// rules in order to produce documentation (CLI help text and markdown) about
+// the lint rules.
+
+#ifndef VERIBLE_VERILOG_ANALYSIS_EXTRACTORS_H_
+#define VERIBLE_VERILOG_ANALYSIS_EXTRACTORS_H_
+
+#include <string>
+#include <set>
+
+#include "absl/status/status.h"
+#include "absl/strings/string_view.h"
+
+namespace verilog {
+namespace analysis {
+
+// Collect all identifiers under module header subtree in the CTS.
+// This could be useful when interface names are required to be
+// preserved.
+absl::Status CollectInterfaceNames(absl::string_view content,
```

Comment:
Suggested test cases for analysis_extractors_test.cc:

* non-modules (function, class, ...) -- should yield empty set
* minimalist module
* module with import declarations in header
* module with parameters
* module with ports
* module with some body statements, to show that none of those identifiers get included in the resulting set
* one source file with two+ modules

Optional test cases:
* nested modules (should internal modules be obfuscated?) -- or punt with TODO

Reply:
Since the `CollectInterfaceNames` extractor should collect names from all modules (including nested ones), the inner module exclusion/inclusion should be an obfuscator issue. I've added a TODO note about this to the obfuscator source file. Is this OK?

Reply:
Yes, this is ok.